### PR TITLE
Product function of AMarketObject class

### PR DESCRIPTION
**Pull request by Njord201**
I added product function of AMarketObject class.

The function returns the number of flames produced since it was last called.

### CODE REVIEW

File: include/Abstract/AMarketObject.hpp [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 #include "Interfaces/IMarketObject.hpp"
```
Comment (Marius-P1) [inline include/Abstract/AMarketObject.hpp:11]:
```suggestion
```

**Review by mathieurobert1 — DISMISSED** [review]
Seems good

File: include/Interfaces/IMarketObject.hpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ class FlameClicker::IMarketObject {
          *
          * @return int - quantity of the object.
```
Comment (Marius-P1) [inline include/Interfaces/IMarketObject.hpp:57]: Update the comment :wink:

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by AxelF44 — DISMISSED** [review]
Job Done

**Review by mathieurobert1 — APPROVED** [review]

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**